### PR TITLE
Fix to missing scheme handling (defaults to http)

### PR DESCRIPTION
**Pull request by jonathon-love**
Fixes r-lib/httr#619
Fixes r-lib/httr#607

### CODE REVIEW

**Comment by cderv** [comment]
Thanks ! 
Two questions I have on this: 

I am just wondering if we should assume that any parse url without scheme should be `http`. why not `https` ? 

Also, I wonder if this type of fix (if scheme is null, assume http) should be better in `request_perform` directly. 
I see `parse_url` as a utility function that helps get the pieces that compose a url. Currently, It could be used to know if a url as a scheme or not by testing if it returns null - with this fix, it can't. Do we want this change in `httr::parse_url`... 🤔 

Just sharing thoughts to help decide what is best.

**Comment by jonathon-love** [comment]
> I am just wondering if we should assume that any parse url without scheme should be `http`. why not `https` ?

the standard behaviour of web-browsers et al. is to default to http when no scheme is provided. and it would be consistent with the earlier versions of `httr` built against curl 3. (additionally, most https sites issue a redirect from `http://` to `https://` anyway)

> Also, I wonder if this type of fix (if scheme is null, assume http) should be better in request_perform directly.

oh yeah, that's a fair point. i'll make that change.

with thanks

File: R/request.R [hunk 1]
```diff
@@ -157,6 +157,9 @@ request_perform <- function(req, handle, refresh = TRUE) {
   }
 
   url_scheme <- parse_url(resp$url)$scheme
+  if (is.null(url_scheme))
```
Comment (hadley) [inline R/request.R:160]: Can you please use `{}` as in https://style.tidyverse.org/syntax.html#indenting?

**Comment by petermeissner** [comment]
I have an [issue](https://github.com/ropensci/robotstxt/issues/50) reported in {robotstxt} directly related to this PR and also solved by this PR.

Will this PR be included in httr at some time in the future?

**Comment by hadley** [comment]
We don't have any plans to spin off a httr release at the moment.

**Comment by petermeissner** [comment]
Thanks, for the heads up.

**Comment by hadley** [comment]
httr has been superseded in favour of httr2, so is no longer under active development. Thanks for using httr, thanks for contributing, and my apologies that your contribution never made it into the package.

**Pull request closed without merging**